### PR TITLE
Adding Transfer Reversals list

### PR DESCRIPTION
**Pull request by jt000**
The [transfer_object](https://stripe.com/docs/api#transfer_object) contains a list of reversals along with reversed and amount_reversed properties. Adding these to the Stripe.NET api

### CODE REVIEW

**Comment by jt000** [comment]
Accidentally merged other changes into this branch... Need to fix (temporarily closing PR)

**Comment by jt000** [comment]
Resolves #313

File: src/Stripe.Tests/transfers/test_data/stripe_transfer_create_options.cs [hunk 1]
```diff
@@ -5,13 +5,13 @@ namespace Stripe.Tests.test_data
 {
     public class stripe_transfer_create_options
     {
-        public static StripeTransferCreateOptions Valid()
+        public static StripeTransferCreateOptions Valid(string recipient = "self")
```
Comment (jaymedavis) [inline src/Stripe.Tests/transfers/test_data/stripe_transfer_create_options.cs]: What are the other options here? This should probably be a static class in the Types folder to follow the convention.

**Comment by jaymedavis** [comment]
This will need to be rebased and readme changed updated. I can handle this in the next release if you don't have time.

**Comment by jaymedavis** [comment]
few small changes:
1) renamed Transfer to TransferId (so it matches BalanceTransactionId). both of these will need to be expandable properties at some point, but there's no reason that has to block this PR.
2) moved the new tests from when_listing_transfers to when_creating_a_failed_transfer
3) instead of the amount 50 being hardcoded in the test, I made it *test_data.stripe_transfer_create_options.Valid(_recipientId).Amount*

Ignore the AppVeyor message above, that was something I was trying but decided to punt on for now. Since the test are not mocked and actually hit stripe, I would have to publish my secret test key in order to get the tests to all pass. That doesn't seem like a good idea.
 
Anyway, this is done and it will be in the next release. I'm going to try and get more stuff completed before I push it up, I'm a bit behind.

Thanks for the help!

**Pull request closed without merging**